app/src/main/java/org/systers/mentorship/view/fragments/RelationPagerFragment.kt

```diff
@@ -17,13 +24,47 @@ class RelationPagerFragment : BaseFragment() {
         fun newInstance() = RelationPagerFragment()
     }
 
+    private lateinit var relationViewModel: RelationViewModel
+    private val activityCast by lazy { activity as MainActivity }
+
     override fun getLayoutResourceId(): Int {
         return R.layout.fragment_relation
     }
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
-        vpMentorshipRelation.adapter = RelationPagerAdapter(childFragmentManager)
-        tlMentorshipRelation.setupWithViewPager(vpMentorshipRelation)
+
+        relationViewModel = ViewModelProviders.of(this).get(RelationViewModel::class.java)
+        relationViewModel.successfulGet.observe(this, Observer {
+            successfull ->
+            activityCast.hideProgressDialog()
+            if (successfull != null) {
+                if (successfull) {
+                    updateView(relationViewModel.mentorshipRelation)
+                } else {
+                    view?.let {
+                        Snackbar.make(it, relationViewModel.message, Snackbar.LENGTH_LONG).show()
+                    }
+                }
+            }
+        })
+
+        activityCast.showProgressDialog(getString(R.string.fetching_users))
```

Comment:
Ups I know you just copy and pasted, so this wasn't you. The message is wrong, you see? the dialog will say that the app is fetching users, can you change to say its fetching/getting current relation information, please?

Reply:
@mohak1283 can you address this please ^^